### PR TITLE
Final shared framework versions

### PR DESCRIPTION
**Pull request by eerhardt**
@brthor @piotrpMSFT @ellismg - let's use this PR to take the stable packages of everything into the CLI.

I think we still need #3529 in here.... /cc @schellap

### CODE REVIEW

**Comment by schellap** [comment]
#3529 already takes most of these changes because of the need to use @brthor's deb fix. It is currently blocked on dotnet-compile-fsc targeting a pre-release version of the NETCore.App. I've given @eerhardt and @brthor push permissions.

**Comment by brthor** [comment]
I've brought all of #3529 over here.

**Comment by brthor** [comment]
@schellap I cannot push to your fork

**Comment by brthor** [comment]
brought in ubuntu 16 changes here as well, this is a huge branch

File: NuGet.Config [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <packageSources>
     <!--To inherit the global NuGet package sources remove the <clear/> line below -->
     <clear />
+    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
```
Comment (ellismg) [inline NuGet.Config]: We should back this out, I guess, since we are trying to push the updated packages to cli-deps.

Comment (eerhardt) [inline NuGet.Config]: Good call.

Comment (eerhardt) [inline NuGet.Config]: Let's see green first....

**Comment by schellap** [comment]
Yay! Some greens. @brthor, thanks, did you accept the invite? :)

**Comment by brthor** [comment]
```
FS0193: internal error: Could not load type 'System.Reflection.PortableExecutable.ContentId' from assembly 'System.Reflection.Metadata, Version=1.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
```

File: src/dotnet/commands/dotnet-new/FSharp_Console/project.json.template [hunk 2]
```diff
@@ -14,7 +14,7 @@
   },
   "tools": {
     "dotnet-compile-fsc": {
-      "version": "1.0.0-*",
+      "version": "1.0.0-preview2-*",
```
Comment (eerhardt) [inline src/dotnet/commands/dotnet-new/FSharp_Console/project.json.template]: necessary?

Comment (brthor) [inline src/dotnet/commands/dotnet-new/FSharp_Console/project.json.template]: yeah there's rc2 versions on the feed. This is probably broken without this

**Comment by eerhardt** [comment]
@dotnet-bot test this please

we now have the "golden" Shared FX build

**Comment by piotrpMSFT** [comment]
FYI https://github.com/dotnet/cli/issues/3558

**Comment by piotrpMSFT** [comment]
@eerhardt note that Windows is passing, as expected.

**Comment by eerhardt** [comment]
Merging.

**Pull request merged**